cmd/promtool/archive.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"os"
+)
+
+const filePerm = 0644
+
+type archiverConfig struct {
+	archiveName string
+}
+
+type archiver interface {
+	Write(fileName string, buf *bytes.Buffer) error
+	Close() error
+	File() *os.File
+}
+
+func newArchiver(cfg archiverConfig) (archiver, error) {
+	file, err := os.Create(cfg.archiveName)
+	if err != nil {
+		return nil, fmt.Errorf("error of creating archive %s: %s", cfg.archiveName, err)
```

Comment:
"error of creating" -> "error creating"

Prefer `%q` over `%s` for file names.